### ISSUE TITLE
`cargo build` cannot build the project

### ISSUE THREAD

**Issue by viirya** [apache/arrow-datafusion#531]
**Describe the bug**
Cannot build the project from a clean checkout.

**To Reproduce**

1. git clone the project
2. cargo build

Then the error:

```
   ...
   Compiling serde_urlencoded v0.7.0
   Compiling h2 v0.3.3
   Compiling tower v0.4.7
   Compiling csv v1.1.6
error: failed to run custom build command for `etcd-client v0.6.3`

Caused by:
  process didn't exit successfully: `/projects/arrow-datafusion/target/debug/build/etcd-client-a91f6504da36f75f/build-script-build` (exit code: 1)
  --- stdout
  cargo:rerun-if-changed=proto
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Expected behavior**

`cargo build` should build the project according to the development document.

**Additional context**
MacOS 11.3

rustc 1.51.0
cargo 1.51.0

**Comment by viirya** [comment]
Reinstalled rust using rustup. Still see:

```
   ...
   Compiling arrow-flight v4.2.0
   Compiling ballista-core v0.5.0-SNAPSHOT (/Users/viirya/repos/arrow-datafusion/ballista/rust/core)
   Compiling etcd-client v0.6.3
error: failed to run custom build command for `ballista-core v0.5.0-SNAPSHOT (/Users/viirya/repos/arrow-datafusion/ballista/rust/core)`

Caused by:
  process didn't exit successfully: `/projects/arrow-datafusion/target/debug/build/ballista-core-9ab32c5dcad9ff02/build-script-build` (exit code: 1)
  --- stdout
  cargo:rerun-if-env-changed=FORCE_REBUILD
  cargo:rerun-if-changed=proto/ballista.proto
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by alamb** [comment]
There is a custom build script in ballista
https://github.com/apache/arrow-datafusion/blob/master/ballista/rust/core/build.rs

I wonder if this is related to needing `protobuf` installed?

@andygrove any thoughts?

**Comment by viirya** [comment]
Oh, thanks for asking. I have `protobuf` installed.

**Comment by alamb** [comment]
🤔

**Comment by edrevo** [comment]
Do you have the required libs installed? See https://github.com/apache/arrow-datafusion/blob/master/ballista/docs/dev-env.md

TL;DR

```
sudo apt-get install -y pkg-config libssl-dev cmake
```

**Comment by viirya** [comment]
On MacOS, I have cmake, pkg-config, openssl (bundled with dev) installed.

**Comment by viirya** [comment]
I uninstalled stable toolchain and install nightly toolchain using `rustup`. And it can build the project.

Still don't know what the issue here because no descriptive error (tried verbose, etc.) can be found. But at lease I can play with this project now.
